### ISSUE TITLE
Planned additions/changes for v1.0

### ISSUE THREAD

**Issue by burodepeper** [source-foundry/compare-typefaces#1]
- [ ] Select a solid set of typefaces to compare
- [x] Add option for comparing regular, italic, bold and bold italic; variants are rendered in a horizontal grid (c0ece56)
- [x] Add option for selecting sizes (b13a097)
- [ ] Use webfonts if a local font is not found
- [ ] Hide unavailable fonts
- [x] Add option to choose a color scheme; at minimal a switch between light and dark variants (1d9ce44)
- [x] Add presets for often confused characters (8d859ac08884d793bd02f914ece7511d72fd9644)
- [x] Display fontName next to sample (f528fbc)
- [ ] Add checkboxes to show/hide a sample
- [x] Use localStorage to store all settings (a762d2ef6c515fc965968b19984b95a0e3a11284)
- [ ] Build Javascript and CSS instead of static files
- [ ] Let users add their own (local) fonts
- [x] Display fontSize for each row in waterfall (8bc13d1)
- [ ] Add option for user to define the order of fonts in the list
- [x] Hold shift when clicking a preset to append text rather than replace (6abd015430e962ad3b88d2e4cea37613beb76693)
- [ ] Allow settings/config override via url parameters, ie:<br>`?variants=all&sizes=9,11,13,15&text=whatever`

**Comment by iandoug** [comment]
Show size for each row in waterfall.

**Comment by burodepeper** [comment]
> Show size for each row in waterfall.

Done.

**Comment by chrissimpkins** [comment]
> Add option to choose a color scheme; at minimal a switch between light and dark variants

👍

**Comment by chrissimpkins** [comment]
Looks fantastic.  Suggest keeping fonts in alphabetical order.  Should be general purpose tool for anyone to use, not Hack specific.

**Comment by burodepeper** [comment]
> Suggest keeping fonts in alphabetical order. Should be general purpose tool for anyone to use, not Hack specific.

I don't (entirely) agree. The user should be able to define the order. Besides that, the first dozen or so fonts should be most used or significant to the purpose of the tool. For example, Courier New shouldn't be near the top of the list just because it starts with a C. However, it should be because it is different from the rest, and as such its design decisions are worth comparing.

**Comment by chrissimpkins** [comment]
> The user should be able to define the order. 

Agree with this because IMO the tool/we should not define those design considerations/priorities.  Could and likely would be designer/dev/project dependent.  Simply a suggestion to expand beyond Hack considerations to general consideration of monospaced faces.

**Comment by iandoug** [comment]
Looking good. Maybe specify that the sizes are px rather than pts?
Unless I really need new glasses :-)

**Comment by burodepeper** [comment]
> Simply a suggestion to expand beyond Hack considerations to general consideration of monospaced faces.

I agree, but focus will be on Hack until it's nearing some sort of v1.0

**Comment by burodepeper** [comment]
> Maybe specify that the sizes are px rather than pts?

Done.

**Comment by iandoug** [comment]
Thanks. Input is rendering as Times, on my program I had to specify it as 
Input Mono
Don't know if this is going to be a platform-specific issue...

font-family (stack): Input
Font being rendered: System Default
font-size: 14px

**Comment by burodepeper** [comment]
@iandoug

> Thanks. Input is rendering as Times, on my program I had to specify it as Input Mono

Will look into these things later. I think it is safe to define font-stacks instead of font-families to deal with all platform specific naming. That would even work with webfonts.

**Comment by burodepeper** [comment]
I changed quite a few things, and wouldn't mind some feedback. You might need to do a hard refresh to get the latest stylesheets and scripts. https://source-foundry.github.io/compare-typefaces/index.html

- Switch between six different color schemes; high, medium and low contrast options, both dark and light
- Choose presets for often confused sets of characters; shift+click appends the characters to your current sample text
- User settings are saved between sessions
- New interface, feels a bit more like a proper tool now

![schermafbeelding 2017-07-13 om 15 20 17](https://user-images.githubusercontent.com/3703530/28168287-e37def16-67de-11e7-91fe-555e26c4ed99.png)

**Comment by burodepeper** [comment]
Haha, I'm already 'having fun' with the tool. I was writing a description for the README and decided to play out a fictional scenario of some confusion in a hospital, and found the following. Compare how numbers blend in with capitals in Hack, compared to Monofur.

The style of the zero could arguably be part of the identity of the typeface, especially to a non-nerd. In Monofur however, the numbers are part of a visually distinct group of glyphs. I am not a fan of the aesthetics, but I have to admit that Monofur has an interesting approach here that you usually only see in 'normal' typefaces.

![schermafbeelding 2017-07-13 om 15 43 15](https://user-images.githubusercontent.com/3703530/28169258-090bcc3c-67e2-11e7-80f6-749eda5fa6bf.png)

![schermafbeelding 2017-07-13 om 15 43 25](https://user-images.githubusercontent.com/3703530/28169263-0d1a6fe0-67e2-11e7-8075-b8afebebbaac.png)

**Comment by iandoug** [comment]
Looks very good. Please add the URL to this discussion.
If you want to make the text a dropdown, here's a list of pangrams that I compiled for keyboard evaluations:
[pangrams.txt](https://github.com/source-foundry/compare-typefaces/files/1145731/pangrams.txt)

Source is a copy of what was originally Wikipedia's list, plus assorted other ones added from various font sites.

**Comment by burodepeper** [comment]
I don't have plans to make the text a dropdown yet, and thanks for the list of pangrams, it could be a nice addition at some point, but I anticipate a different kind of sample at the moment. I have a list somewhere with some collections of glyphs that make sense together.

**Comment by chrissimpkins** [comment]
This looks absolutely fantastic David!  

> Compare how numbers blend in with capitals in Hack, compared to Monofur.

We are going to be able to recognize alot of new issues that were never apparent before with this tool.  Something that I would suggest that falls along the lines of the pangram specimens suggested by @iandoug are source code specimens so that we can see how some of these issues play out in real code.  

Now we need to come up with an approach to prioritize what is important and should be addressed...

**Comment by burodepeper** [comment]
Merci, merci! : )

> Now we need to come up with an approach to prioritize what is important and should be addressed...

We could start by passively going through old, open issues, and examining those with this tool. We need to learn how to extract useful information from our tools, so we can address issues with well founded arguments. My philosophy is that solving problems isn't a community effort, but that finding issues is. Once you have a solid suggestion for a solution, it is valuable to present it to the community to detect any holes in it you might have missed.

Potential new issues --issues that aren't as obvious like the one regarding monofur-- and especially those that aren't urgent, probably need even more analysis before you throw them out into the wild. For instance something that bugs me, is that smileys, such as `:-)` don't line up nicely. Of course, the problem isn't with the smiley as such, but rather with how punctuation is positioned relative to, in this case, the x-height. I doubt it is smart to start open discussions about such things, without a solid case on which to base whatever arguments. After all, this is the internet, where everyone is an expert.

But most importantly, I like playing with my new tool ; )

**Comment by chrissimpkins** [comment]
> We could start by passively going through old, open issues, and examining those with this tool

All legibility issues are currently in the legible repository issue reports.  I went through every open issue report in Hack and consolidated them there.  Tagged with labels for type of issue.  Let's use the threads there for discussion.  That was the intent!

> But most importantly, I like playing with my new tool ; )

👍

**Comment by iandoug** [comment]
Observation: I find the font names a bit difficult to read due to their small size.

Decided to see how this string would work as a test... root 2 with some zeroes and ones replaced with O and l. Some fonts make you wonder which is the O and which is the 0.

1.4142l3562373O95048801688724209698O78569671875376948073l76

(pi doesn't have enough zeroes in the first few dozen digits. Though I suppose a manually-made random string would also do...)

**Comment by chrissimpkins** [comment]
poke!  

ready to start using this to examine glyphs?
